Adapters/Vungle/ISVungleAdapter/src/main/java/com/ironsource/adapters/vungle/VungleBannerAdapter.java

```diff
@@ -0,0 +1,204 @@
+
+    VungleBannerAdapter(String placementId, ISBannerSize size, AdConfig adConfig, BannerSmashListener listener) {
+        this.mListener = listener;
+        this.mISBannerSize = size;
+
+        if (adConfig == null) {
+            adConfig = new AdConfig();
+        }
+
+        mBannerAd = new BannerAd(ContextProvider.getInstance().getApplicationContext(), placementId, adConfig);
+        mBannerAd.setAdListener(this);
+    }
+
+    void load() {
+        mBannerAd.load();
+    }
+
+    public void loadWithBid(String serverData) {
+        mBannerAd.load(serverData);
+    }
+
+    public void destroy() {
+        if (mBannerAd != null) {
+            mBannerAd.finishAd();
+            mBannerAd = null;
+        }
+        mListener = null;
+        mISBannerSize = null;
+    }
+
+    @Override
+    public void adLoaded(@NonNull BaseAd baseAd) {
+        IronLog.ADAPTER_CALLBACK.verbose("placementId = " + baseAd.getPlacementId());
+
+        mAdLoaded = true;
+
+        if (mListener == null) {
+            IronLog.INTERNAL.verbose("listener is null");
+            return;
+        }
+
+        if (mISBannerSize == null) {
+            IronLog.INTERNAL.verbose("banner size is null");
+            return;
+        }
+
+        if (!mBannerAd.canPlayAd()) {
+            IronLog.ADAPTER_CALLBACK.error("can't play ad");
+            mListener.onBannerAdLoadFailed(ErrorBuilder.buildLoadFailedError("can't play ad"));
+            return;
+        }
+
+        BannerView bannerView = mBannerAd.getBannerView();
+
+        if (bannerView != null) {
+            mListener.onBannerAdLoaded(bannerView, getBannerLayoutParams(mISBannerSize));
+        } else {
+            IronLog.ADAPTER_CALLBACK.error("banner view is null");
+            mListener.onBannerAdLoadFailed(ErrorBuilder.buildLoadFailedError("Vungle LoadBanner failed - banner view is null"));
+        }
+    }
+
+    @Override
+    public void adStart(@NonNull BaseAd baseAd) {
+        // no-op
+    }
+
+    @Override
+    public void adImpression(@NonNull BaseAd baseAd) {
+        IronLog.ADAPTER_CALLBACK.verbose("placementId = " + baseAd.getPlacementId());
+
+        if (mListener == null) {
+            IronLog.INTERNAL.verbose("listener is null");
+            return;
+        }
+
+        mListener.onBannerAdShown();
+    }
+
+    @Override
+    public void adClick(@NonNull BaseAd baseAd) {
+        IronLog.ADAPTER_CALLBACK.verbose("placementId = " + baseAd.getPlacementId());
+
+        if (mListener == null) {
+            IronLog.INTERNAL.verbose("listener is null");
+            return;
+        }
+
+        mListener.onBannerAdClicked();
+    }
+
+    @Override
+    public void adEnd(@NonNull BaseAd baseAd) {
+        // no-op
+    }
+
+    @Override
+    public void error(@NonNull BaseAd baseAd, @NonNull VungleException exception) {
+        if (mAdLoaded) {
+            IronLog.ADAPTER_CALLBACK.verbose("placementId = " + baseAd.getPlacementId() + ", exception = " + exception);
```

Comment:
Is there no playback error callback for banner presentation?

Reply:
we have, but not exported separately.